src/vs/platform/terminal/common/capabilities/bufferMarkCapability.ts

```diff
@@ -0,0 +1,53 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Emitter } from 'vs/base/common/event';
+import { IBufferMarkDetectionCapability, TerminalCapability } from 'vs/platform/terminal/common/capabilities/capabilities';
+import { IMarkProperties } from 'vs/platform/terminal/common/terminalProcess';
+// Importing types is safe in any layer
+// eslint-disable-next-line code-import-patterns
+import type { IMarker, Terminal } from 'xterm-headless';
+
+/**
+ * Stores marks to be added to the buffer
+ * anonymous marks are those that have no ID
+ */
```

Comment:
"Anonymous marks" is a concept internal to the class so it shouldn't be mentioned in the jsdoc as it would then show up in intellisense externally.

```suggestion
/**
 * Manages "marks" in the buffer which are lines that are tracked when lines are added to or removed
 * from the buffer.
 */
```